index.html

```diff
@@ -1012,6 +1012,158 @@ <h3>
+          </li>
+          <li>[=list/For each=] |entry:ordered map| of |json|["file_handlers"]:
+            <ol>
+              <li>Let |file_handler:ordered map| be [=process a file handler
+              item=] with |entry|.
+              </li>
+              <li>If |file_handler| is failure, continue.
+              </li>
+              <li>[=list/Append=] |file_handler| to |processedFileHandlers|.
+              </li>
+            </ol>
+          </li>
+        </ol>
+        <p>
+          On [=installation=], a user agent SHOULD register the file handlers
+          with the operating system via interactions that are consistent with:
+        </p>
+        <ul>
+          <li>The [=file handler/icon=] and [=file handler/name=] appear in the
+          host operating system surfaces presenting options for opening files,
+          for example within "Open with..." context menus for associated file
+          types.
+          </li>
+          <li>The [=file handler/icon=] appears as the icon of associated files
+          in the host operating system.
+          </li>
+          <li>File types with multiple possible handlers can be disambiguated
+          by UX provided by the operating system.
+          </li>
+        </ul>
+        <p>
+          A user agent MAY prevent registered file handlers from registering as
+          the 'default' file handler for a given type if the operating system
+          allows.
+        </p>
+        <section data-cite="file-system-access">
+          <h2>
+            <dfn>Execute a file handler launch</dfn>
+          </h2>
+          <p>
+            This is the IDL used for executing a file handling launch:
+          </p>
+          <pre class="idl">
+            partial interface Window {
+              readonly attribute LaunchQueue launchQueue;
+            };
+
+            callback LaunchConsumer = any (LaunchParams params);
+
+            [Exposed=Window] interface LaunchQueue {
+              undefined setConsumer(LaunchConsumer consumer);
+            };
+
+            [Exposed=Window] interface LaunchParams {
+              readonly attribute FrozenArray&lt;FileSystemHandle&gt; files;
+            };
+          </pre>
+          <p>
+            When a [=file type=] is registered with a web app and one or more
+            of these registered files are launched on the platform, run the
+            following steps, given the launched {{FrozenArray}} of
+            {{FileSystemHandle}}s |files:FrozenArray&lt;FileSystemHandle&gt;|
+            and the corresponding [=file handler|file_handler=].
+          </p>
+          <ol>
+            <li>Let |url| be the |file_handler|.[=file handler/action=].
+            </li>
+            <li>Let |browsing context| be the result of creating a new
+            [=top-level browsing context=].
+            </li>
+            <li>[=Navigate=] |browsing context| to |url|.
+            </li>
+            <li>XXX Wait for navigation to be complete?
```

Comment:
We do wait for the navigation to "complete", whatever that means technically.

Currently we also have to wait for the user to approve the action (at least sometimes).

However, we're interested in changing this such that user approval, if any, occurs before the creation of the browsing context and subsequent navigation. This sidesteps concerns like what happens after a navigation has occurred, but the user has not allowed the open-file action. It sounds like you're describing this potential future improvement rather than the current state of Chrome. Is that accurate?

Reply:
I added some notes about what we might want to do here instead - let me know what you think.

Reply:
PTAL